### ISSUE TITLE
REM: Refactor generate_inverse_confusion_matrix

### ISSUE THREAD

**Issue by amirebrahimi** [unitaryfund/mitiq#1562]
Per [PR feedback](https://github.com/unitaryfund/mitiq/pull/1449#discussion_r972863821) it was suggested to replace the use of the `NoisyReadoutSampler` and `cirq.measure_confusion_matrix` with a fixed analytical solution. Additionally, Revert changes from #1605.

As mentioned in the PR it is okay if this is a follow-up PR to #1449.

Here's a [reference](https://github.com/quantumlib/Cirq/pull/4854/files#diff-6037b8997ce87fcfcfe512167b87b8fb3b420061e5fd835f2fb2f3ce64554dd2R22)

**Comment by Misty-W** [comment]
Hi @amirebrahimi, should we include this issue in milestone 0.21? For context, the target completion date of milestone 0.21 is Nov 30.

**Comment by amirebrahimi** [comment]
I think so as it should be a relatively small PR. #1449 needs to land first.

**Comment by amirebrahimi** [comment]
Would you please assign this to me as well, @Misty-W? Thanks!
